src/Features/Core/Portable/SpellCheck/AbstractSpellCheckSpanService.cs

```diff
@@ -0,0 +1,173 @@
+                SyntaxToken token,
+                CancellationToken cancellationToken)
+            {
+                ProcessTriviaList(token.LeadingTrivia, cancellationToken);
+
+                if (_syntaxFacts.IsStringLiteral(token) ||
+                    token.RawKind == _syntaxKinds.SingleLineRawStringLiteralToken ||
+                    token.RawKind == _syntaxKinds.MultiLineRawStringLiteralToken)
+                {
+                    AddSpan(new SpellCheckSpan(token.Span, SpellCheckKind.String));
+                }
+                else if (token.RawKind == _syntaxKinds.InterpolatedStringTextToken &&
+                         token.Parent?.RawKind == _syntaxKinds.InterpolatedStringText)
+                {
+                    AddSpan(new SpellCheckSpan(token.Span, SpellCheckKind.String));
+                }
+                else if (token.RawKind == _syntaxKinds.IdentifierToken)
+                {
+                    TryAddSpanForIdentifier(token);
+                }
+
+                ProcessTriviaList(token.TrailingTrivia, cancellationToken);
+            }
+
+            private void TryAddSpanForIdentifier(SyntaxToken token)
+            {
+                // Leverage syntactic classification which already has to determine if an identifier token is the name of
+                // some construct.
+                var classification = _classifier.GetSyntacticClassificationForIdentifier(token);
+                switch (classification)
+                {
+                    case ClassificationTypeNames.ClassName:
+                    case ClassificationTypeNames.RecordClassName:
+                    case ClassificationTypeNames.DelegateName:
+                    case ClassificationTypeNames.EnumName:
+                    case ClassificationTypeNames.InterfaceName:
+                    case ClassificationTypeNames.ModuleName:
+                    case ClassificationTypeNames.StructName:
+                    case ClassificationTypeNames.RecordStructName:
+                    case ClassificationTypeNames.TypeParameterName:
+                    case ClassificationTypeNames.FieldName:
+                    case ClassificationTypeNames.EnumMemberName:
+                    case ClassificationTypeNames.ConstantName:
+                    case ClassificationTypeNames.LocalName:
+                    case ClassificationTypeNames.ParameterName:
+                    case ClassificationTypeNames.MethodName:
+                    case ClassificationTypeNames.ExtensionMethodName:
+                    case ClassificationTypeNames.PropertyName:
+                    case ClassificationTypeNames.EventName:
+                    case ClassificationTypeNames.NamespaceName:
+                    case ClassificationTypeNames.LabelName:
+                        break;
+                    default:
+                        return;
+                }
+
+                AddSpan(new SpellCheckSpan(token.Span, SpellCheckKind.Identifier));
```

Comment:
this flow was slightly confusing - can this just be moved into where the break is (and add return)?

Reply:
yup.  done.